pkg/types/types.go

```diff
@@ -10,34 +14,40 @@ const (
 	StateReady      = "ready"
 
 	// labels
-	HarvesterNodeLabelKey   = "harvesterhci.io/managed"
-	HarvesterNodeLabelValue = "true"
-	SupportBundleLabelKey   = "harvesterhci.io/supportbundle"
-	DrainKey                = "kubevirt.io/drain"
-
-	AppManager = "support-bundle-manager"
-	AppAgent   = "support-bundle-agent"
+	SupportBundleLabelKey       = "rancher/supportbundle"
+	SupportBundleNodeLabelValue = "true"
+	DrainKey                    = "kubevirt.io/drain"
 
-	BundleCreationTimeout = 8 * time.Minute
-	NodeBundleWaitTimeout = "5m"
+	SupportBundleManager = "support-bundle-manager"
+	SupportBundleAgent   = "support-bundle-agent"
 )
 
 type ManagerStatus struct {
-	// phase to collect bundle
-	Phase string
-
-	// fail to collect bundle
-	Error bool
-
-	// error message
+	Phase        string
+	Error        bool
 	ErrorMessage string
+	Progress     int
+	FileName     string
+	FileSize     int64
+}
+
+type SupportBundle struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// progress of the bundle collecting. 0 - 100.
-	Progress int
+	Spec   SupportBundleSpec   `json:"spec,omitempty"`
+	Status SupportBundleStatus `json:"status,omitempty"`
+}
 
-	// bundle filename
-	Filename string
+type SupportBundleSpec struct {
+	IssueURL    string `json:"issueURL"`
+	Description string `json:"description"`
+}
 
-	// bundle filesize
-	Filesize int64
+type SupportBundleStatus struct {
+	State      string                   `json:"state,omitempty"`
+	Progress   int                      `json:"progress,omitempty"`
+	FileName   string                   `json:"filename,omitempty"`
+	FileSize   int64                    `json:"filesize,omitempty"`
+	Conditions []harversterv1.Condition `json:"conditions,omitempty"`
```

Comment:
We don't need to use `harvesterv1.Condition` ? but we can remove this dependency later.

Reply:
@yasker fixed